iac-extensions/arm/src/main/java/org/sonar/iac/arm/checks/utils/CheckUtils.java

```diff
@@ -50,4 +57,29 @@ public static Predicate<Expression> isTrue() {
   public static Predicate<Expression> isFalse() {
     return expr -> expr.is(ArmTree.Kind.BOOLEAN_LITERAL) && !((BooleanLiteral) expr).value();
   }
+
+  public static Predicate<Expression> isEmptyString() {
+    return expr -> TextUtils.isValue(expr, "").isTrue();
+  }
+
+  public static Predicate<Expression> isArrayWithValues() {
+    return expr -> expr.is(ArmTree.Kind.ARRAY_EXPRESSION) && !((ArrayExpression) expr).elements().isEmpty();
+  }
+
+  public static Predicate<Expression> isEmptyArray() {
+    return expr -> expr.is(ArmTree.Kind.ARRAY_EXPRESSION) && ((ArrayExpression) expr).elements().isEmpty();
+  }
+
+  public static Predicate<Expression> isEmptyObject() {
+    return expr -> expr.is(ArmTree.Kind.OBJECT_EXPRESSION) && ((ObjectExpression) expr).properties().isEmpty();
+  }
+
+  public static Predicate<Expression> isFunctionCall(String functionName) {
+    // TODO SONARIAC-10038 ARM Json: parse expression in string and build the AST to be same as Bicep equivalent
+    // Here we detect functionCall in two ways:
+    // - in Bicep we expect a FunctionCall object
+    // - in Json we expect a StringLiteral with this format: "[functionName(...)]"
+    return expr -> (expr.is(ArmTree.Kind.STRING_LITERAL) && isRegexMatch("^\\[" + functionName + "\\(.*\\)\\]$").test(expr)) // ARM Json
```

Comment:
Does it allow spacing after `[` or before `]`?

Reply:
That's a good question.
I tried to decompile some json file into bicep, the result is that your string must expressly start with `[` and end with `]`.
Otherwise, the expression in the brackets can contain spaces anywhere.
The examples I tried:
```
"[uniqueString('secret')]"    -> uniqueString('secret')
"   [uniqueString('secret')]" -> '   [uniqueString(\'secret\')]'
"[uniqueString('secret')]   " -> '[uniqueString(\'secret\')]   '
"[   uniqueString('secret')]" -> uniqueString('secret')
"[uniqueString('secret')   ]" -> uniqueString('secret')
"[uniqueString(   'secret')]" -> uniqueString('secret')
"[uniqueString('secret'   )]" -> uniqueString('secret')
```
I will adapt the regex to support this 👍